packages/pwa-kit-extension-sdk/TUTORIAL.md

```diff
@@ -0,0 +1,404 @@
+3. **Granular Control**: You can override specific components or pages while keeping the rest of the extension's functionality intact.
+
+#### Implementing Your First Override
+
+Let's customize the home page by creating an override. First, we'll establish the necessary directory structure:
+
+Here, we will be customizing the home page by using **Extension Overrides.** It is a customization mechanism that works at the **file level.** Each extension can declare a list of “overridable” files. If a file is overridable, you can create a file in the “overrides” folder to replace the implementation.
+
+* Places all overrides under the `/app/overrides` directory
+* include the full extension package name as path
+* Mirrors the internal path of the file being overridden
+
+```sh
+# Create a folder called overrides, in the app folder
+# This is where you will keep all extension overrides
+mkdir -p ./app/overrides/@salesforce/extension-chakra-storefront/pages/home/
+
+touch ./app/overrides/@salesforce/extension-chakra-storefront/pages/home/index.jsx
+```
+
+Now, update your `index.jsx` with custom implementation:
+
+```js
+import React from 'react'
+
+const Home = () => {
+    return <h1>Hello World</h1>
+}
+
+export default Home
+```
+
+For now, you need to restart the dev server for new override files to take effect. Once the server is started, navigate to home page and you should see
+#### Important Considerations
+
+🙋 Note: After creating new override files, you'll need to restart your development server for the changes to take effect. This is because override resolution happens during the application's bootstrap phase.
+
+⚠️ Warning: Not every file in an extension is overridable. Extension developers carefully choose which files to make overridable to maintain a stable public interface. This deliberate approach helps:
+
+* Prevent breaking changes when updating extensions
+* Maintain clear boundaries between extension and custom code
+* Ensure reliable upgrade paths for your application
+
+#### Override Discovery and Documentation
+
+To discover which files are overridable in an extension:
+
+1. Check the extension's documentation for a list of overridable files
+2. Look for files imported with the `overridable!` prefix within the extension
+
+In the next exercise, we'll explore how to create your own extension and make its components overridable...
+
+## Exercise #2. Create Your Extension
+
+After mastering the use of existing extensions, you're ready to create your own. In this exercise, we'll explore how to build a custom extension that can be shared across projects or even published for others to use. Let's dive into the development process that will transform you from an extension consumer to an extension author.
+
+### Getting Started with Extension Development
+
+First, we’ll generate a fresh PWA Kit project by running the following command:
+
+```sh
+npx @salesforce/pwa-kit-create-app@4.0.0-extensibility-preview.3
```

Comment:
nit: Should we use a NPM tag instead like `extensibility-preview-latest` and point that tag to the latest extensibility preview instead of using `4.0.0-extensibility-preview.3`?

We probably will do at least one more release fixing bugs before beta so the version will most likely `4.0.0-extensibility-preview.4`

Reply:
Good call, let's create the tag first and then we can update the docs.